src/software/new_geom/ray.h

```diff
@@ -6,91 +6,74 @@ class Ray final
 {
    public:
     /**
-     * Creates a degenerate Ray at (0, 0)
+     * Creates a Ray starting at (0, 0) with zero angle
```

Comment:
Could we change this to 
``` suggestion
     * Creates a Ray starting at (0, 0) with a degenerate direction vector (ie. (0,0))
```
Also we aren't actually doing this in the default constructor? We should explicitly set these values to zero there.

Reply:
Done

Reply:
Oh so we're storing the direction as an angle, so setting it to a direction of "zero" isn't degenerate right? We probably want to say something like "... at (0,0) pointing along the positive x-axis"

Reply:
Sorry about that, misunderstood how we stored direction earlier.

Reply:
Fixed! Yeah direction of "zero" is just the zero angle.